pkg/entpgx/pgxpool_dialect_driver.go

```diff
@@ -0,0 +1,258 @@
+	return &EntPgxpoolDriver{
+		pool: pool,
+	}
+}
+
+type EntPgxpoolDriver struct {
+	pool pool.PGXConnPool
+}
+
+func (e *EntPgxpoolDriver) Exec(ctx context.Context, query string, args, result any) error {
+	argv, ok := args.([]any)
+	if !ok {
+		return fmt.Errorf("dialect/sql: invalid type %T. expect []any for args", result)
+	}
+	switch result := result.(type) {
+	case nil:
+		if _, err := e.pool.Exec(ctx, query, argv...); err != nil {
+			return err
+		}
+	case *sql.Result:
+		commandTag, err := e.pool.Exec(ctx, query, argv...)
+		if err != nil {
+			return err
+		}
+		*result = execResult{rowsAffected: commandTag.RowsAffected()}
+	default:
+		return fmt.Errorf("dialect/sql: invalid type %T. expect *sql.Result", result)
+	}
+	return nil
+}
+
+func (e *EntPgxpoolDriver) Query(ctx context.Context, query string, args, v any) error {
+	vr, ok := v.(*sql.Rows)
+	if !ok {
+		return fmt.Errorf("dialect/sql: invalid type %T. expect *sql.Rows", v)
+	}
+	argv, ok := args.([]any)
+	if !ok {
+		return fmt.Errorf("dialect/sql: invalid type %T. expect []any for args", args)
+	}
+	pgxRows, err := e.pool.Query(ctx, query, argv...)
+	if err != nil {
+		return err
+	}
+	columnScanner := &entPgxRows{pgxRows: pgxRows}
+	*vr = sql.Rows{
+		columnScanner,
+	}
+	return nil
+}
+
+func (e *EntPgxpoolDriver) ExecContext(ctx context.Context, query string, args ...any) (stdsql.Result, error) {
+	commandTag, err := e.pool.Exec(ctx, query, args...)
+	if err != nil {
+		return nil, err
+	}
+	return &execResult{rowsAffected: commandTag.RowsAffected()}, nil
+}
+
+func (e *EntPgxpoolDriver) Tx(ctx context.Context) (dialect.Tx, error) {
+	return e.BeginTx(ctx, nil)
+}
+
+func (e *EntPgxpoolDriver) BeginTx(ctx context.Context, opts *sql.TxOptions) (dialect.Tx, error) {
+	pgxOpts, err := getPgxTxOptions(opts)
+	if err != nil {
+		return nil, err
+	}
+	tx, err := e.pool.BeginTx(ctx, *pgxOpts)
+	if err != nil {
+		return nil, err
+	}
+	return &EntPgxPoolTx{
+		tx: tx,
+	}, nil
+}
+
+func getPgxTxOptions(opts *sql.TxOptions) (*pgx.TxOptions, error) {
+	var pgxOpts pgx.TxOptions
+	if opts == nil {
+		return &pgxOpts, nil
+	}
+	switch opts.Isolation {
+	case stdsql.LevelDefault:
+	case stdsql.LevelReadUncommitted:
+		pgxOpts.IsoLevel = pgx.ReadUncommitted
+	case stdsql.LevelReadCommitted:
+		pgxOpts.IsoLevel = pgx.ReadCommitted
+	case stdsql.LevelRepeatableRead, stdsql.LevelSnapshot:
+		pgxOpts.IsoLevel = pgx.RepeatableRead
+	case stdsql.LevelSerializable:
+		pgxOpts.IsoLevel = pgx.Serializable
+	default:
+		return nil, fmt.Errorf("unsupported isolation: %v", opts.Isolation)
+	}
+
+	if opts.ReadOnly {
+		pgxOpts.AccessMode = pgx.ReadOnly
+	}
+	return &pgxOpts, nil
+}
+
+func (e *EntPgxpoolDriver) Close() error {
+	e.pool.Close()
+	return nil
+}
+
+func (e *EntPgxpoolDriver) Dialect() string {
+	return dialect.Postgres
+}
+
+type EntPgxPoolTx struct {
+	tx pgx.Tx
+}
+
+func (e *EntPgxPoolTx) Exec(ctx context.Context, query string, args, result any) error {
+
```

Comment:
nit: Unnecessary new line, some linters will even complain about this... probably should add a linter to this repo.

There's a few other similar instances of this.